src/main/java/com/conveyal/r5/analyst/cluster/GridResultAssembler.java

```diff
@@ -0,0 +1,201 @@
+
+        try {
+            File gzipFile = File.createTempFile(request.jobId, ".access_grid.gz");
+
+            buffer.close();
+
+            // There's probably a more elegant way to do this with NIO and without closing the buffer
+            InputStream is = new BufferedInputStream(new FileInputStream(temporaryFile));
+            OutputStream os = new GZIPOutputStream(new BufferedOutputStream(new FileOutputStream(gzipFile)));
+            ByteStreams.copy(is, os);
+            is.close();
+            os.close();
+
+            LOG.info("GZIP compression reduced regional analysis {} from {}mb to {}mb ({}x compression)",
+                    request.jobId,
+                    temporaryFile.length() / 1024 / 1024,
+                    gzipFile.length() / 1024 / 1024,
+                    temporaryFile.length() / gzipFile.length()
+                    );
+
+            s3.putObject(outputBucket, String.format("%s.access", request.jobId), gzipFile);
+
+            // these will fill up the disk lickety-split if we don't keep them cleaned up.
+            gzipFile.delete();
+            temporaryFile.delete();
+        } catch (Exception e) {
+            LOG.error("Error uploading results of regional analysis {}", request.jobId, e);
+        }
+    }
+
+    /** Process an SQS message */
+    public void handleMessage (Message message) {
+        try {
+            byte[] body = base64.decode(message.getBody());
+            ByteArrayInputStream bais = new ByteArrayInputStream(body);
+            LittleEndianDataInputStream data = new LittleEndianDataInputStream(bais);
+
+            // ensure that it starts with ORIGIN
+            char[] header = new char[6];
+            for (int i = 0; i < 6; i++) {
+                header[i] = (char) data.readByte();
+            }
+
+            if (!"ORIGIN".equals(new String(header))) {
+                error = true;
+                LOG.error("Origin not in proper format for query %s");
+                return;
+            }
+
+            int version = data.readInt();
+
+            if (version != ORIGIN_VERSION) {
+                error = true;
+                LOG.error("Origin version mismatch for query %s, expected {}, found {}", ORIGIN_VERSION, version);
+            }
+
+            int x = data.readInt();
+            int y = data.readInt();
+            int iterations = data.readInt();
+
+            // convert the remainder to a delta coded byte array
+            ByteArrayOutputStream originData = new ByteArrayOutputStream(iterations * 4);
+            LittleEndianDataOutputStream origin = new LittleEndianDataOutputStream(originData);
+            for (int i = 0, prev = 0; i < iterations; i++) {
+                int current = data.readInt();
+                origin.writeInt(current - prev);
+                prev = current;
+            }
+            origin.close();
+
+            // write to the buffer in a synchronized block to ensure no threading issues
```

Comment:
"Write to the proper subregion of the unified results buffer for this origin number."